### PR TITLE
Background editor

### PR DESCRIPTION
**Pull request by on21viktoria**
added functionalities to select picture and background color for postcard

### CODE REVIEW

File: src/components/Editor/Sidebar/Background.vue [hunk 1]
```diff
@@ -39,4 +58,5 @@ export default Vue.extend({
 .tool-title.subtitle {
   font-size: 16px;
 }
```
Comment (ON21MB) [inline src/components/Editor/Sidebar/Background.vue:60]: There are two closing </style>-tags.

File: src/components/Editor/Sidebar/Sticker.vue [hunk 2]
```diff
@@ -1,42 +1,198 @@
...
-      Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus
-      ac facilisis in, egestas eget quam. Morbi leo risus, porta ac consectetur
-      ac, vestibulum at eros.
-    </p>
-    <b-img
-      src="https://picsum.photos/500/500/?image=54"
-      fluid
-      thumbnail
-    ></b-img>
     <!-- <p class="tool-title subtitle">Suche</p>
```
Comment (ON21MB) [inline src/components/Editor/Sidebar/Sticker.vue:4]: Do we still need this code which is toggled?

File: src/components/Editor/PostcardLayout.vue [hunk 3]
```diff
@@ -45,14 +43,19 @@
...
+      style="color: #fff"
+      >Speichern & Zum Warenkorb</router-link
+    >
   </div>
 </template>
 
 <script lang="ts">
 import { EventBus } from "@/main";
 import Vue from "vue";
+import { mapGetters, mapState } from "vuex";
```
Comment (ON21MB) [inline src/components/Editor/PostcardLayout.vue:58]: This import doesn't work in my VS.

File: src/components/Editor/Sidebar/Photos.vue [hunk 4]
```diff
@@ -29,19 +506,22 @@ import Vue from "vue";
 
 export default Vue.extend({
   name: "Photos",
-  components: {},
-  methods: {},
+  methods: {
+    changeBackgroundPicture(picture: string) {
+      this.$store.dispatch("setPicture", picture);
```
Comment (ON21MB) [inline src/components/Editor/Sidebar/Photos.vue:511]: This method doesn't work in my VS, it has problems with the .$store part.

File: src/components/Editor/Sidebar/Sticker.vue [hunk 5]
```diff
@@ -1,42 +1,198 @@
...
 <script lang="ts">
 import Vue from "vue";
 
 export default Vue.extend({
   name: "Sticker",
-  components: {},
-  props: {
+  methods: {
+    changeSticker(sticker: string){
+      this.$store.dispatch("setSticker", sticker)
```
Comment (ON21MB) [inline src/components/Editor/Sidebar/Sticker.vue:142]: Same here, the method doesn't work.

File: src/store/pictures.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+export default new Vuex.Store({
+  state: {
+    currentPicture: "images-trvl-card/LargeLetterDesign_Berlin.jpg",
+    currentBackgroundColor: "#ffffff",
+    currentSticker: "",
+  },
+  mutations: {
+    setCurrentPicture(state, payload) {
+      state.currentPicture = payload;
```
Comment (ON21MB) [inline src/store/pictures.ts:15]: In my VS, there are problems with the type of state.

**Review by ON21MB — CHANGES_REQUESTED** [review]
I think there are a few things that need to be checked before merging the branch. I can't load the page so far! I'll review it once again after the changes!

File: src/components/Editor/Sidebar/Background.vue [hunk 7]
```diff
@@ -1,34 +1,53 @@
...
 export default Vue.extend({
   name: "Background",
-  components: {},
-  props: {
+  methods: {
+    changeColor() {
+      let colorselector = document.getElementById(
+        "selection-color"
+      ) as HTMLInputElement;
+      this.$store.dispatch("setBackgroundColor", colorselector.value);
```
[the remark this answers is not in the document]
Comment (on21viktoria) [reply src/components/Editor/Sidebar/Background.vue:31]: einmal npm install machen

**Comment by ON21MB** [comment]
**I tried the branch again after running npm install. I could visit the page!**

- The color picker for the background works fine. 
- It's just a bit confusing because there's written "you can choose from either a background picture or a color" but actually in this section of the toolbar you can only choose the color - the photos are in Robin's "photo"-part of the sidebar, aren't they? I'd consider calling the section "background-color" and is it maybe possible to resize the color-preview-box? Because for now it looks a bit "broken", from a user's point of view it's not really clear to me that I have to click on the small white box to change the background-color.
- Also applying the stickers to the postcard works well. 
- the this.$store.dispatch()-method is still underlined in red in my VS, but I'm not sure if this is just a problem of VS, since it works. 
- If we pay attention that all other comments have been resolved, for me we can merge the branch!

**Pull request merged**